src/player.js

```diff
@@ -1770,9 +1796,50 @@ export default class Player extends FakeEventTarget {
   _updateTracks(tracks: Array<Track>): void {
     Player._logger.debug('Tracks changed', tracks);
     this._tracks = tracks;
+    this._maybeAddExternalTracks();
     this._addTextTrackOffOption();
   }
 
+  /**
+   * checking if there is external caption to add. if so -  adding it.
+   * @returns {void}
+   * @private
+   */
+  _maybeAddExternalTracks(): void {
+    if (this._config.sources.captions) {
+      this._externalCaptionsHandler.addTracks();
+    }
+  }
+
+  /**
+   * adds a new text track element to the video element or set an existing one
+   * (when adding a text track with existing language to the video element it will remove all its cues)
+   * @param {PKTextTrack} textTrack - the playkit text track object to be added
+   * @returns {void}
+   * @public
+   */
+  addOrSetNativeTextTrack(textTrack: textTrack): void {
+    const _engineTextTrack = this._engine.addOrSetTextTrack(textTrack);
+    // safari always push the text track at the beginning, so we have to update the index of the indexes in the player
+    // text track module.
+    if (_engineTextTrack && this._engine.indexOfSameLanguageTrack === -1) {
```

Comment:
this not justifying new engine API.
Check it by getTracks(TrackType.TEXT) and iterating all text tracks.